packages/scope-manager/src/referencer/Referencer.ts

```diff
@@ -521,10 +521,15 @@ class Referencer extends Visitor {
   protected JSXOpeningElement(node: TSESTree.JSXOpeningElement): void {
     this.referenceJsxPragma();
     if (node.name.type === AST_NODE_TYPES.JSXIdentifier) {
-      if (node.name.name[0].toUpperCase() === node.name.name[0]) {
+      if (
+        node.name.name[0].toUpperCase() === node.name.name[0] ||
+        node.name.name === 'this'
+      ) {
```

Comment:
But in order to be analysed the logic needs to be outside the JSXIdentifier. For example if we move the logic from JSXOpeningElement, JSXIdentifier would never be visited/analysed. 

Besides, the JSXIdentifier is agnostic whether it is being analyzed from a JSXMemberExpression or a JSXOpeningElement, so I cannot access these information inside the selector making it difficult to manage the logic inside the selector.

Sorry if I didn't understand your proposal correctly.